GitHubRepositories.ps1

```diff
@@ -191,6 +191,125 @@ function New-GitHubRepository
 }
 
+function New-GitHubRepositoryFromTemplate
+{
+<#
+    .SYNOPSIS
+        Creates a new repository on GitHub from a template repository.
+
+    .DESCRIPTION
+        Creates a new repository on GitHub from a template repository.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER RepositoryName
+        Name of the repository to be created.
+
+    .PARAMETER TemplateOwnerName
+        Owner of the template repository.
+
+    .PARAMETER TemplateRepositoryName
+        Name of the template repository.
+
+    .PARAMETER OwnerName
+        Owner of the repository to be created. If not specified, the DefaultOwnerName configuration
+        property value will be used.
+
+    .PARAMETER Description
+        A short description of the repository.
+
+    .PARAMETER Private
+        By default, this repository will created Public.  Specify this to create a private
+        repository.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        New-GitHubRepositoryFromTemplate -RepositoryName MyNewRepo -TemplateOwnerName MyOrg -TemplateRepositoryName MyTemplateRepo -OwnerName Me
+
+        Creates a new GitHub repository from the specified template repository
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+        )]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "",
+        Justification="Methods called within here make use of PSShouldProcess, and the switch is
+        passed on to them inherently.")]
+    param(
+        [Parameter(
+            Mandatory,
+            Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TemplateOwnerName,
+
+        [Parameter(
+            Mandatory,
+            Position = 3)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TemplateRepositoryName,
+
+        [string] $OwnerName,
+
+        [string] $Description,
+
+        [switch] $Private,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements -BoundParameters $PSBoundParameters
+    $OwnerName = $elements.ownerName
+
+    $telemetryProperties = @{
+        RepositoryName = (Get-PiiSafeString -PlainText $RepositoryName)
```

Comment:
I'd probably capture the source template repo name/owner as well (properly redacted).